internal/pkg/cli/app_deploy.go

```diff
@@ -0,0 +1,379 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cli
+
+import (
+	"errors"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/archer"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/aws/session"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/build/docker"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/build/ecr"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/manifest"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/store/ssm"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/prompt"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/workspace"
+)
+
+// BuildAppDeployCommand builds the `app deploy` subcommand.
+func BuildAppDeployCommand() *cobra.Command {
+	input := &appDeployOpts{
+		globalOpts: newGlobalOpts(),
+		prompt:     prompt.New(),
+	}
+
+	cmd := &cobra.Command{
+		Use:  "deploy",
+		Long: `Deploy an application to an environment.`,
+		Example: `
+  Deploy an application named "frontend" to a "test" environment.
+  $ archer app deploy --name frontend --env test`,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if err := input.init(); err != nil {
+				return err
+			}
+
+			if err := input.sourceInputs(); err != nil {
+				return err
+			}
+
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := input.deployApp(); err != nil {
+				return err
+			}
+
+			return nil
+		},
+		// PostRunE: func(cmd *cobra.Command, args []string) error {
+		// TODO: recommended actions?
+		// },
+	}
+
+	cmd.Flags().StringVarP(&input.app, "app", "a", "", "application name")
+	cmd.Flags().StringVarP(&input.env, "env", "e", "", "environment name")
+	cmd.Flags().StringVarP(&input.imageTag, "tag", "t", "", "image tag")
+
+	return cmd
+}
+
+type appDeployOpts struct {
```

Comment:
Can `appDeployOpts` implement the `actionCommand` interface? this becomes useful when we want to embed appDeploy to the "archer init" cmd

https://github.com/aws/amazon-ecs-cli-v2/blob/e7fea8c28349c3c223e6d3f3c541c95a1bb58a5e/internal/pkg/cli/cli.go#L31-L36